app/assets/javascripts/discourse-markdown-it/src/features/bbcode-block.js

```diff
@@ -1,4 +1,24 @@
-let isWhiteSpace;
+let isWhiteSpace, escapeHtml;
+
+function camelCaseToDash(str) {
+  return str.replace(/([a-zA-Z])(?=[A-Z])/g, "$1-").toLowerCase();
+}
+
+export function applyDataAttributes(token, attributes, defaultName) {
```

Comment:
This method was extracted from `d-wrap` and slightly changed. In order to prevent flaky tests, the keys of the `attributes` we want to convert into `data-*` attributes are not always added in alphabetical order, instead of the order they were added in the hash.